test/mailers/mailer_test.rb

```diff
@@ -0,0 +1,20 @@
+require "test_helper"
+
+class MailerTest < ActionMailer::TestCase
+  test "pass given block to #mail call" do
+    class TestMailer < Devise::Mailer
+      def confirmation_instructions(record, token, opts = {})
+        @token = token
+        devise_mail(record, :confirmation_instructions, opts) do |format|
+          format.html(content_transfer_encoding: "7bit")
+        end
+      end
+    end
+
+    Devise.mailer = TestMailer
```

Comment:
@lucasmazza we don't need ``deliver_now`` in this case (it also fails on Rails 4.1, because it's ``deliver`` there).
``confirmation_instructions`` returns the mail object we want to test.